lib/resources/widgets/text_form_field_for_dob.dart

```diff
@@ -0,0 +1,75 @@
+import 'package:application_form/resources/colors/app_color.dart';
+import 'package:application_form/resources/extensions/extension.dart';
+import 'package:flutter/material.dart';
+
+import '../constants/constants.dart';
+import '../global_variables/global_variable.dart';
+
+class TextFormFieldForDob extends StatelessWidget {
+  const TextFormFieldForDob({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    final bodySmall = Theme.of(context).textTheme.bodySmall;
+    final bodyLarge = Theme.of(context).textTheme.bodyLarge;
+    return Column(
+      crossAxisAlignment: CrossAxisAlignment.start,
+      children: [
+        Text(
+          "When were you born?",
+          style: bodyLarge,
+        ),
+        SizedBoxConstants.sizedBoxHeightEight,
+        TextFormField(
+          textInputAction: TextInputAction.next,
+          controller: dobController,
+          decoration: InputDecoration(
+            hintText: "Date of Birth*",
+            suffixIcon: Icon(
+              Icons.calendar_today_outlined,
+              color: AppColors.datePicker.iconColor,
+              size: DoubleConstants.datePickerIconSize,
+            ),
+          ),
+          keyboardType: TextInputType.none,
+          style: bodySmall,
+          autovalidateMode: AutovalidateMode.onUserInteraction,
+          onTap: () async {
+            final DateTime? date = await showDatePicker(
+              context: context,
+              initialDate: DateTime.now(),
+              firstDate: DateTime(IntConstants.startDate),
+              lastDate: DateTime(IntConstants.endDate),
+              locale: const Locale('en', 'GB'),
+              builder: (context, child) => Theme(
+                data: ThemeData().copyWith(
+                    colorScheme: ColorScheme.light(
+                      primary: AppColors.datePicker.color,
```

Comment:
The border color of the initial date, it is the same as the primary color. But it should be black in color. I tried changing color values of different parameters of color schema and also tried searching on internet but did not found an answer to it.

Reply:
Here is the code for the widget implementation https://github.com/flutter/flutter/blob/20fa32e129bd9b01f60f6152467c6e534eeb5e0d/packages/flutter/lib/src/material/calendar_date_picker.dart#L936. Can you tell me now why you weren't able to solve the issue. Also, can you suggest alternative approach.

Reply:
The color for the border of the initial date cannot be changed because it takes the color that has been assigned to the primary parameter. This is how it is designed. So, it's color cannot be changed.

Reply:
@Sujey-Naik Not accepting this answer. Please explain why this is not possible.

Reply:
There is a primary parameter for the theme data of the date picker in the color scheme. So this primary parameter takes a color value. The color that is assigned to the primary value is taken by the header part of the calendar and also the date that is selected is of primary color. The border of initial date also uses the same primary color. So, I cannot change it to a different color. When I try to change the color for primary parameter, to make the border color of initial date change, the color assigned to selected date and header of the calendar also changes. So it is not possible to make changes in the border color of initial date.